### ISSUE TITLE
Could not load type 'System.Security.Cryptography.HMACRIPEMD160' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.

### ISSUE THREAD

**Issue by gobenaus** [darkoperator/Posh-SSH#393]
Hi,

We were trying to use the library on Ubuntu and when we try to establish a new session to an SFTP server we get the following error:

Could not load type 'System.Security.Cryptography.HMACRIPEMD160' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.

So it must have an issue loading the SSH.NET library under the hood somewhere I would think. Has anyone run into this before? Do we need to recompile the library against .Net core. 

Thanks in advance for any pointers.

**Comment by darkoperator** [comment]
Which version? The Betas are the only ones Im testing support for .Net Core that would run in Linux

**Comment by gobenaus** [comment]
I saw the same issue raised earlier https://github.com/darkoperator/Posh-SSH/issues/217 and at the time Powershell Core was not supported. It seems to indicate to support Powershell 7.0 now, so just wanted to confirm whether it should work on Core or not. Thanks

**Comment by gobenaus** [comment]
We are using 2.3.0 not the beta release. Powershell 7.1.3 and Posh 2.3.0

**Comment by darkoperator** [comment]
Only the beta versions should work on linux, that version is Windows only

**Comment by gobenaus** [comment]
Ok. We'll give the betas a try. Thank you.

**Comment by gobenaus** [comment]
We did test 3.0 beta and it works on Linux. I do have one question. There serve serveral cmdLet names in 2.x that ended in file, such as Get-SFTPFile. There is also a more generic Get-SFTPItem which exists in both 2.x and 3.0. Will the Get-SFTPFile cmdLets be re-added to 3.0 or should we start updating our code to use the more generic names. Thanks

**Comment by gobenaus** [comment]
Release notes say the cmdLets were removed, so I guess they won't be in 3.0. Should have checked notes before posting.

**Comment by gobenaus** [comment]
Hi,

The assembly load error is gone. We only tested it on Linux as this is the only place we are using version 7.x. So we no longer have the original issue. We just noted that a few cmdLets had been removed in that version and that was already noted in the release notes. Since version 2.x also supports the Verb-SFTPItem cmdLets we can just update our code and not reference the Verb-SFTPFile cmdLets. I assume they were the same in 2.x and just had two names for the same thing.

**Comment by darkoperator** [comment]
Yep I left them in 2 introduced the new ones and in the redme and changelog before the change in 3 when I release it
